src/systems/ec/implementations/react/packages/checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { queryOne } from '@ecl/ec-base/helpers/dom';
+
+import Icon from '@ecl/ec-react-component-icon/Icon';
+
+const Checkbox = ({
+  className,
+  defaultChecked,
+  disabled,
+  id,
+  helperText,
+  hideLabel,
+  invalid,
+  invalidText,
+  label,
+  name,
+  removeError,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-checkbox', {
+    'ecl-checkbox--invalid': invalid,
+  });
+
+  return (
+    <div className="ecl-form-group ecl-form-group--checkbox">
+      <div className={classNames}>
+        <input
+          {...props}
+          type="checkbox"
+          id={id}
+          defaultChecked={defaultChecked}
+          name={name || undefined}
+          className="ecl-checkbox__input"
+          disabled={disabled}
+          onChange={removeError}
+        />
+
+        <label
+          className={classnames(
+            'ecl-form-label ecl-form-label--inline ecl-checkbox__label',
+            {
+              'ecl-form-label--invalid': invalid,
+              'ecl-checkbox__label--hidden': hideLabel,
+              'ecl-form-label--hidden': hideLabel,
+            }
+          )}
+          htmlFor={id}
+        >
+          <span className="ecl-checkbox__box">
+            <Icon className="ecl-checkbox__icon" shape="ui--check" size="xs" />
+          </span>
+          {label}
+        </label>
+      </div>
+
+      {invalid && invalidText && (
+        <div className="ecl-feedback-message">{invalidText}</div>
+      )}
+      {helperText && <div className="ecl-help-block">{helperText}</div>}
+    </div>
+  );
+};
+
+Checkbox.propTypes = {
+  className: PropTypes.string,
+  defaultChecked: PropTypes.bool,
+  disabled: PropTypes.bool,
+  helperText: PropTypes.node,
+  hideLabel: PropTypes.bool,
+  id: PropTypes.string.isRequired,
+  invalid: PropTypes.bool,
+  invalidText: PropTypes.node,
+  label: PropTypes.string,
+  name: PropTypes.string,
+  removeError: PropTypes.func,
+};
+
+Checkbox.defaultProps = {
+  className: '',
+  defaultChecked: false,
+  disabled: false,
+  helperText: '',
+  hideLabel: false,
+  invalid: false,
+  invalidText: '',
+  label: '',
+  name: '',
+  removeError: () => {
+    const inPage = queryOne('.ecl-checkbox--invalid');
```

Comment:
Eh, it is just a matter of being able to display in the demo the style specifications for a checkbox with errors.
Before i was preventing the input from being checked (to avoid the inconsistent styling that it would have been creating), after the romain's review i decided to apply this simple js to make the component usable in the website and in storybook.
I absolutely agree with you, this is here just to properly demo this version of the component.